src/server/accounting.c

```diff
@@ -2259,3 +2279,51 @@ void log_alter_records_for_attrs(job *pjob, svrattrl *plist) {
 	if (entire_record[0] != '\0')
 		account_record(PBS_ACCT_ALTER, pjob, entire_record);
 }
+
+
+/**
+ * @brief
+ * Common function to log a suspend/resume record 
+ * for suspend/resume job events respectively.
+ * 
+ * @param[in] pjob - job to log records for.
+ * @param[in] user - User who requested the signal suspend/resume 
+ * @param[in] host - Hostname where the signal was triggered
+ * @param[in] acct_type - Accounting type flag
+ * 
+ * @returns void 
+ */
+void
+log_suspend_resume_record(job *pjob, char *user, char *host, int acct_type)
+{
+	if (acct_type == PBS_ACCT_SUSPEND) {
+		char *resc_buf;
+		int resc_buf_size = RESC_USED_BUF_SIZE;
+
+		/* Allocating initial space as required by resc_used. Future space will be allocated by pbs_strcat(). */
+		resc_buf = malloc(RESC_USED_BUF_SIZE);
+		if (resc_buf == NULL)
+			return;
+
+		resc_buf[0] = '\0';
+
+		if (get_resc_used(pjob, resc_buf, resc_buf_size) < 0) {
+			write_account_record(acct_type, pjob->ji_qs.ji_jobid, NULL);
+			goto end;
```

Comment:
Labels and goto are considered error prone/bad practice (https://stackoverflow.com/questions/3517726/what-is-wrong-with-using-goto) unless used in a very obvious way, like at the end of the function as an error label. In this case, your label is inside an inner block, so I'd recommend not using it. You actually are not saving any extra lines of code anyways, your label is 3 lines long including the label, and it saves exactly 3 free(resc_buf); lines, so it's not valuable either. So, please remove it.